src/core/segment_reader.rs

```diff
@@ -149,6 +150,21 @@ impl SegmentReader {
         }
     }
 
+    /// Accessor to the `BytesFastFieldReader` associated to a given `Field`.
+    /// May panic if the field is not a Bytes fast field.
+    pub fn bytes_fast_field_reader(&self, field: Field) -> fastfield::Result<BytesFastFieldReader> {
+        let field_entry = self.schema.get_field_entry(field);
```

Comment:
maybe test if the type is right now that you have the field entry, and return the error here.

The benefit is then that you `panic!` only if the index is corrupted, but you return a nice error if the field just does not have the right type.

Reply:
Should I turn FastFieldNotAvailableError into a FastFieldError enum to indicate the fast field is the wrong type? Just saying it's not available seems confusing.

Reply:
ok. I also think making the two unindistiguishable is not the best but we can keep it that way for the moment.